app/parse.py

```diff
@@ -2,11 +1,156 @@
+import csv
+import time
+
+import requests
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from dataclasses import dataclass, fields
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://mate.academy"
+
+
+class Driver:
+    instance = None
+
+    @staticmethod
+    def initialize() -> webdriver.Chrome:
+        if Driver.instance is None:
+            Driver.instance = webdriver.Chrome()
+        return Driver.instance
 
 
 @dataclass
 class Course:
     name: str
     short_description: str
     duration: str
+    num_topics: str | None = None
+    num_modules: str | None = None
+
+
+def parse_single_course(soup: BeautifulSoup) -> Course:
+    name = soup.select_one(".ProfessionCard_title__Zq5ZY").text
```

Comment:
This line (and others similar to it) uses a magic string as a CSS selector. Magic strings are strings that have a special meaning in your code, but that meaning is not immediately obvious from the string itself. It's a good practice to avoid magic strings, as they can make your code harder to understand and maintain. Consider defining these CSS selectors as constants at the top of your file or in a separate configuration file. This way, you can give them meaningful names, and you only need to update them in one place if they ever change.